### PR TITLE
Fix the numpy Apple M1 build

### PR DESCRIPTION
**Pull request by kulinseth**
There was a build issue on numpy with M1:
AssertionError: would build wheel with unsupported tag ('cp39', 'cp39', 'macosx_11_0_universal2')

The issue was fixed in the packaging 20.5 and onwards, but that was not
getting picked by setuptools. So explicitly adding this dependency in
the toml file. The change was tested with:

```
python3 -m pip install --force-reinstall .
```

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 [build-system]
 # Minimum requirements for the build system to execute.
 requires = [
+    "packaging==20.5",
```
Comment (mattip) [inline pyproject.toml:4]: I wonder if we should signal that this is needed for M1, what do you think? I think `arm64` is the correct one, `aarch64` is for linux.
```suggestion
    "packaging==20.5; platform_machine=='arm64'",
```

Comment (kulinseth) [reply pyproject.toml:4]: Makes sense.
Do you know what might be going on with "Build_Test/full" failure.

**Comment by mattip** [comment]
Did you verify this is correct? We don't build/test for M1 here

**Comment by kulinseth** [comment]
> Did you verify this is correct? We don't build/test for M1 here

Yes, I used `python3 -m pip install --force-reinstall .` to check and it builds fine on M1.

**Comment by kulinseth** [comment]
Some of the tests are failing with:
```
 File "/opt/python/cp38-cp38/lib/python3.8/sqlite3/dbapi2.py", line 27, in <module>
    from _sqlite3 import *
ModuleNotFoundError: No module named '_sqlite3'
```
@mattip How to find out if this is related to this change ? Any help would be appreciated.

**Comment by charris** [comment]
> Some of the tests are failing with:

It's a bug in the manylinux2010 image released last Wednesday. Fixed in NumPy by pinning manylinux2010 to the previous version, and now fixed upstream as well.

**Comment by mattip** [comment]
Thanks @kulinseth

**Pull request merged**